File: app/src/main/java/com/example/counter/MainActivity.java
package com.example.counter;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.google.android.material.chip.Chip;

public class MainActivity extends AppCompatActivity {

    private int NowCount = 0;
    private int todayCount = 0;
    private int lifetimeCount = 0;



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);

        SharedPreferences preferences = getSharedPreferences("com.example.counter", Context.MODE_PRIVATE);
        boolean isSwitch1On = preferences.getBoolean("switch1", false);
        TextView currentCount = findViewById(R.id.current_chip);
        if  (isSwitch1On) {
            currentCount.setVisibility(View.GONE);

        }

        SharedPreferences preference = this.getSharedPreferences("com.example.counter", Context.MODE_PRIVATE);
        if (preference.getBoolean("firstRun", true)) {
            startActivity(new Intent(this, SetupActivity.class));
        }

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;

        });

        @SuppressLint("CutPasteId")

        final Chip currentChip = findViewById(R.id.current_chip);
        final Chip todayChip = findViewById(R.id.today_chip);
        final Chip lifetimeChip = findViewById(R.id.lifetime_chip);
        Button button = findViewById(R.id.button);
        Button setting = findViewById(R.id.cardvieww);

        button.setOnClickListener(new View.OnClickListener() {
            @SuppressLint("SetTextI18n")
            @Override
            public void onClick(View v) {
                if (isSwitch1On) {
                    todayCount++;
                    lifetimeCount++;
                } else {
                    NowCount++;
                    if (NowCount > 108) {
                        NowCount = 1;
                        todayCount++;
                        lifetimeCount++;
                    }
                }
                currentChip.setText("Current Count: " + NowCount);
                todayChip.setText("Today's Count: " + todayCount);
                lifetimeChip.setText("Lifetime Count: " + lifetimeCount);
            }
        });


    setting.setOnClickListener(new View.OnClickListener() {
        @Override
        public void onClick(View v) {
                startActivity(new Intent(MainActivity.this, SetupActivity.class));
        }
    });
    }

}

File: app/src/main/java/com/example/counter/MainActivity.kt
package com.example.counter

import android.annotation.SuppressLint
import android.content.Context
import android.content.Intent
import android.icu.text.SimpleDateFormat
import android.net.Uri
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.LinearLayout
import android.widget.ScrollView
import android.widget.TextView
import android.widget.ToggleButton
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.google.android.material.chip.Chip
import java.util.Date
import java.util.Locale

class MainActivity : AppCompatActivity() {
    private var nowCount = 0
    private var todayCount = 0
    private var lifetimeCount = 0

    @SuppressLint("SetTextI18n", "CutPasteId", "MissingInflatedId")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        this.enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        val txtview7 = findViewById<TextView>(R.id.textView7)

        val sharedPref = getSharedPreferences("myPrefs", Context.MODE_PRIVATE)
        lifetimeCount = sharedPref.getInt("count", 0)

        txtview7.setOnClickListener {
            val url = "https://github.com/Tyagism/Digital_Counter/blob/master/Privacy_Policy.md"
            val intent = Intent(Intent.ACTION_VIEW)
            intent.data = Uri.parse(url)
            startActivity(intent)
        }


        val preferences = getSharedPreferences("com.example.counter", MODE_PRIVATE)
        val isSwitch1On = preferences.getBoolean("switch1", false)
        val currentCount = findViewById<TextView>(R.id.current_chip)
        if (isSwitch1On) {
            currentCount.visibility = View.GONE
        }

        val preference = getSharedPreferences("com.example.counter", MODE_PRIVATE)
        if (preference.getBoolean("firstRun", true)) {
            startActivity(Intent(this, SetupActivity::class.java))
        }
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v: View, insets: WindowInsetsCompat ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
        @SuppressLint("CutPasteId") val currentChip = findViewById<Chip>(R.id.current_chip)
        val todayChip = findViewById<Chip>(R.id.today_chip)
        val lifetimeChip = findViewById<Chip>(R.id.lifetime_chip)
        @SuppressLint("MissingInflatedId", "LocalSuppress")

        val buttn = findViewById<Button>(R.id.buttoncount)
        val toggleButton = findViewById<ToggleButton>(R.id.toggleButton)
        val linearLayout = findViewById<LinearLayout>(R.id.linear_layoutv)
        val histry = findViewById<TextView>(R.id.textView6)
        val scrollView = findViewById<ScrollView>(R.id.scrol)

        var isHistoryVisible = false

        buttn.setOnClickListener {
            if (toggleButton.isChecked) {
                todayCount++
                lifetimeCount++
            } else {
                nowCount++
                if (nowCount > 108) {
                    nowCount = 1
                    todayCount++
                    lifetimeCount++
                }
            }
            currentChip.text = "Current Count : $nowCount"
            todayChip.text = "Today's Count : $todayCount"
            lifetimeChip.text = "Lifetime Count : $lifetimeCount"

            val sharedPref = getSharedPreferences("myPrefs", Context.MODE_PRIVATE)
            val editor = sharedPref.edit()
            editor.putInt("count", lifetimeCount)
            editor.apply()

            val currentDateTime = SimpleDateFormat("yy-MM-dd  hh:mm", Locale.getDefault()).format(Date())

            val newEntry = TextView(this)
            newEntry.text = "Lifetime Count : $lifetimeCount at $currentDateTime /n"

            linearLayout.addView(newEntry)
        }

        toggleButton.setOnCheckedChangeListener { _, isChecked ->
            currentChip.visibility = if (isChecked) View.GONE else View.VISIBLE
        }

        histry.setOnClickListener {
            isHistoryVisible = !isHistoryVisible
            scrollView.visibility = if (isHistoryVisible) View.VISIBLE else View.GONE
        }















    } }

